core/store/src/trie/mod.rs

```diff
@@ -289,26 +289,26 @@ fn decode_children(cursor: &mut Cursor<&[u8]>) -> Result<[Option<CryptoHash>; 16
 }
 
 impl RawTrieNode {
-    fn encode_into(&self, out: &mut Vec<u8>) -> Result<(), std::io::Error> {
-        let mut cursor = Cursor::new(out);
+    fn encode_into(&self, out: &mut Vec<u8>) {
+        out.clear();
```

Comment:
```suggestion
```

Hm, this is a behavioral change for this method, we weren't clearing the buffer before. 

I would also expect this to *not* clear buffer, similarly how to how, eg, https://doc.rust-lang.org/stable/std/io/trait.Read.html#method.read_to_string works. 

If you just want to get a vec with data, that's what `.encode` is for.

Reply:
Thanks, I think I somehow misunderstood how this was being used by one of the callers.